api/src/main/java/org/openmrs/module/openconceptlab/importer/Saver.java

```diff
@@ -665,13 +656,29 @@ private void updateOrAddNamesFromOcl(Concept concept, OclConcept oclConcept) {
 					name.setUuid(version5Uuid(oclConcept.getUrl() + "/names/" + oclName.getUuid()).toString());
 				}
 				name.setConceptNameType(oclNameType);
-				name.setLocalePreferred(oclName.isLocalePreferred());
+
+				// If this concept does not yet have any names preferred in the given locale, make it preferred
+				// This is needed in order to match expected state in the Concept.addName method
+				boolean preferred = oclName.isLocalePreferred();
+				if (!preferred) {
+					preferred = !hasExistingPreferredNameInExactLocale(concept, oclName.getLocale());
+				}
```

Comment:
This is ultimately all that this PR is really about - this is the bug fix I am introducing.  Everything else in this PR is about demonstrating the issue and the fix through unit tests (which necessitate the 2.x profile, etc)

Reply:
Not quite sure I understand what the issue was and why this was the fix?